<?xml version="1.0"?>
<!--
show_all_point_cloud.launch: show all point cloud in rviz
Author: Ravi Joshi
Date: 2018/02/16
-->
<launch>
  <!-- rostopic for receving point cloud -->
  <arg name="pc1_topic" default="/kinect1/qhd/points"/>
  <arg name="pc2_topic" default="/kinect2/qhd/points"/>
  <arg name="pc3_topic" default="/kinect3/qhd/points"/>

  <arg name="pc1_frame_id" default="/kinect1_link"/>
  <arg name="pc2_frame_id" default="/kinect2_link"/>
  <arg name="pc3_frame_id" default="/kinect3_link"/>

  <node name="show_all_point_cloud" pkg="multiple_kinect_baxter_calibration" type="show_all_point_cloud" output="screen">
    <param name="pc1_topic" value="$(arg pc1_topic)"/>
    <param name="pc2_topic" value="$(arg pc2_topic)"/>
    <param name="pc3_topic" value="$(arg pc3_topic)"/>
    <param name="pc1_frame_id" value="$(arg pc1_frame_id)"/>
    <param name="pc2_frame_id" value="$(arg pc2_frame_id)"/>
    <param name="pc3_frame_id" value="$(arg pc3_frame_id)"/>
  </node>

  <include file="$(find multiple_kinect_baxter_calibration)/launch/calibration_publisher.launch">
    <arg name="multiple_kinects" value="true"/>
  </include>

  <node pkg="rviz" type="rviz" respawn="false" name="rviz" args="-d $(find multiple_kinect_baxter_calibration)/files/show_all_pc.rviz"/>
</launch>
